classify-cluster-priority/README.md

```diff
@@ -0,0 +1,85 @@
+# Classifying clusters based on priority
+
+Contents:
+
+1. User stories
+2. Current practice
+3. Proposal
+4. Considerations and explanations
+5. Possible applications
+
+## User stories
+
+- As a cluster admin, I want to easily communicate the purpose and importance of a cluster to those interactig with it, so that they can act accordingly.
+- As Giant Swarm operations staff I want to understand the importance of clusters so I can make appropriate decisions, e. g. when prioritizing problem mitigation in an incident affecting multiple clusters.
+
+## Current practice
+
+- One customer uses the first character of the cluster name to distinguish between production (`p`), development (`d`), and staging `s`. This customer also has clusters which don't fall into either of these categories and their names start wtith arbotrary letters.
+- Many customers use specific terms and phrases in the cluster description, e. g. "Production", "test", "testing", "ignore alerts", "prd", "prod", "nonprod".
+- Some customers use labels on the `clusters.cluster.x-k8s.io` resource (the cluster's main resource) indicating the importance. E. g.
+    - `odp/environment: development`
+    - `odp/environment: staging`
+    - `odp/environment: production`.
+
+## Proposal
+
+We introduce a label and a schema for the label values to indicate the relative importance of a cluster.
+
+Label name: `giantswarm.io/service-priority`
+
+The label is to be used on the main resource defining the cluster, usually of resource type `clusters.cluster.x-k8s.io`.
+
+To simplify maintenance and avoid logical conflicts, this label SHOULD NOT be applied on other resources of the cluster. Systems MUST ignore the label used on other resources in the same cluster. This means that it is not possible, for example, to differentiate between the importance of sevewral node pools within the same cluster.
+
+### Well-defined label values
+
+There are three well-defined label values to be used.
+
+- `giantswarm.io/service-priority: production`: A cluster labelled this way serves production traffic (i. e. end-user facing). Clusters with this label value are considered the most important ones, relative to other clusters not carrying this label.
+
+- `giantswarm.io/service-priority: staging`: Clusters carrying this label are used to simulate running on a production cluster, but are not serving production traffic. They are considered less important than ones carrying the above `production` label.
+
+- `giantswarm.io/service-priority: development`: A cluster with this label is used by developers to develop and test applications. It is considered less important than a cluster labelled as `production` or `staging`.
```

Comment:
I'm not sure these values match the label. They're more environment than priority. Add to this the fast that "staging" has a few dozen different names in different orgs.

I was thinking something like `Critical` (e.g. production sales), `Important` (e.g. CI or job clusters, back office services with failover capabilities), `Low` (e.g. clusters that can afford downtime, especially out of working hours)

Reply:
I agree on Marcus proposal here

Reply:
Agreed. This also would help with the use-case of clusters being prepared. Example: Customer is building up a cluster for production but traffic is not there yet meaning its a production cluster with low priority but could switch label for Critical when actually serving traffic.

Reply:
☝️  That's a very good point. I hadn't thought about that use case. Might be worth explicitly calling it out in the user stories.

Reply:
I agree with the above. 
Another use case supporting the above - a customer may decide their dev cluster has higher importance, since they don't want their developers to be waiting around doing nothing.

Reply:
Nice input, thank you!

Reply:
Coming in with the next commit. I'm resolving this thread.